src/TuxState.js

```diff
@@ -0,0 +1,650 @@
+'use strict';
+
+var assign = require('object-assign');
+var invariant = require('./TuxInvariant');
+
+// var assign = function(obj, extendObj){
+//   for (var key in extendObj) {
+//     if (extendObj.hasOwnProperty(key)) {
+//       obj[key] = extendObj[key];
+//     }
+//   }
+//   return obj;
+// }
+
+/*
+callback = function(newSt, key, stAtKey, valAtKey) {
+  newSt[key] = valAtKey * stAtKey
+  delete newSt[key];
+};
+*/
+
+var buildNewState = function (currentState, newProps, callback) {
+  var newState = assign({}, currentState);
+  var keyChain = [];
+  var recurseKeys = function (currentSt, newSt, newPr) {
+    for (var key in newPr) {
+      if (newPr.hasOwnProperty(key)) {
+        var valAtKey = newPr[key];
+        //error check valAtKey
+        var stAtKey = currentSt[key];
+        if (valAtKey !== null && typeof valAtKey === 'object' && !Array.isArray(valAtKey) && stAtKey !== null && typeof stAtKey === 'object' && !Array.isArray(stAtKey)) {
+          newSt[key] = assign({}, stAtKey);
+          recurseKeys(stAtKey, newSt[key], valAtKey);
+        } else {
+          callback(newSt, key, stAtKey, valAtKey);
+
+        }
+      }
+    }
+  };
+  recurseKeys(currentState, newState, newProps);
+  return newState;
+};
+
+
+var hasAnyOuterKeysMatching = function (currentState, newProps) {
+  for (var key in newProps) {
+    if (currentState[key]) {
+      return true;
+    }
+  }
+  invariant(!newProps, "No keys match any keys in the current state.");
+};
+
+var invariantNumberCheck = function (input) {
+  if (typeof(input) !== "number") {
+    invariant(!input, 'Cannot perform operation on "%s" because it is not of type number.', input);
+  };
+};
+
+var invariantArrayCheck = function (input, message) {
+  if (!Array.isArray(input)) {
+    invariant(!input, 'Cannot perform operation on "%s" because it is not an array.', input);
+  };
+};
+
+var invariantValueCheck = function (input) {
+  if (typeof(input) === "object") {
+    invariant(!input, 'Cannot perform operation on "%s" because it must be a single value.', input);
```

Comment:
Change to `invariant(false, ...)`.  Also change error message to `Cannot perform operation on "%s" because it is an object`.